controllers/usersignup/approval_test.go

```diff
@@ -24,19 +22,10 @@ import (
 
 func TestGetClusterIfApproved(t *testing.T) {
 	// given
+	os.Setenv("WATCH_NAMESPACE", commontest.HostOperatorNs)
```

Comment:
could you please remind me why we need this? 
there is a function that automatically reverts the change, let's use it instead:
```suggestion
	commontest.SetEnvVarAndRestore(t, "WATCH_NAMESPACE", commontest.HostOperatorNs)
```

Reply:
This is what we get without it:

```
    --- FAIL: TestGetClusterIfApproved/with_two_clusters_available,_the_second_one_is_defined_as_the_last-used_one (0.01s)
        toolchainconfig.go:735: WATCH_NAMESPACE env var is not set, defaulting to 'toolchain-host-operator'
        approval_test.go:45: 
                Error Trace:    /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/host-operator/controllers/usersignup/approval_test.go:45
                Error:          Received unexpected error:
                                WATCH_NAMESPACE must be set
                                failed to get watch namespace
                                github.com/codeready-toolchain/toolchain-common/pkg/configuration.LoadLatest
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/toolchain-common/pkg/configuration/cache.go:51
                                github.com/codeready-toolchain/toolchain-common/pkg/configuration.GetConfig
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/toolchain-common/pkg/configuration/cache.go:80
                                github.com/codeready-toolchain/host-operator/controllers/toolchainconfig.GetToolchainConfig
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/host-operator/controllers/toolchainconfig/configuration.go:41
                                github.com/codeready-toolchain/host-operator/controllers/usersignup.getClusterIfApproved
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/host-operator/controllers/usersignup/approval.go:78
                                github.com/codeready-toolchain/host-operator/controllers/usersignup.TestGetClusterIfApproved.func1
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/host-operator/controllers/usersignup/approval_test.go:42
                                testing.tRunner
                                        /home/lkrejci/sdk/go1.22.1/src/testing/testing.go:1689
                                runtime.goexit
                                        /home/lkrejci/sdk/go1.22.1/src/runtime/asm_amd64.s:1695
                                unable to get ToolchainConfig
                                github.com/codeready-toolchain/host-operator/controllers/usersignup.getClusterIfApproved
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/host-operator/controllers/usersignup/approval.go:80
                                github.com/codeready-toolchain/host-operator/controllers/usersignup.TestGetClusterIfApproved.func1
                                        /home/lkrejci/Projects/go/src/github.com/codeready-toolchain/host-operator/controllers/usersignup/approval_test.go:42
                                testing.tRunner
                                        /home/lkrejci/sdk/go1.22.1/src/testing/testing.go:1689
                                runtime.goexit
                                        /home/lkrejci/sdk/go1.22.1/src/runtime/asm_amd64.s:1695
                Test:           TestGetClusterIfApproved/with_two_clusters_available,_the_second_one_is_defined_as_the_last-used_one
```